packages/mdc-list/foundation.js

```diff
@@ -109,29 +112,77 @@ class MDCListFoundation extends MDCFoundation {
 
   /** @param {number} index */
   setSelectedIndex(index) {
-    if (index === this.selectedIndex_) {
-      return;
+    if (index < 0 && index >= this.adapter_.getListItemCount()) return;
+
+    if (this.adapter_.hasCheckboxAtIndex(index)) {
+      this.setAriaAttributesForCheckbox_(index);
+    } else if (this.adapter_.hasRadioAtIndex(index)) {
+      this.setAriaAttributesForRadio_(index);
+    } else {
+      this.setAriaAttributesForSingleSelect_(index);
+      this.setClassNamesForSingleSelect_(index);
     }
 
-    const className = this.useActivatedClass_
-      ? cssClasses.LIST_ITEM_ACTIVATED_CLASS : cssClasses.LIST_ITEM_SELECTED_CLASS;
+    if (this.selectedIndex_ >= 0 && this.selectedIndex_ !== index) {
+      this.adapter_.setAttributeForElementIndex(this.selectedIndex_, 'tabindex', -1);
+    } else if (this.selectedIndex_ === -1 && index !== 0) {
```

Comment:
That's correct. We're saving additional `tabindex` change which will be set in below statement anyway in that case. Added a code comment about this line.

I didn't want to bring additional complexity by allowing `-1` tabindex since this method is now used for single select, checkbox and radio variants which was not the case before.